docs/sources/flow/reference/components/prometheus.exporter.cloudwatch.md

```diff
@@ -0,0 +1,396 @@
+| ------------- | -------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------ | ------- | -------- |
+| `regions`     | `list(string)` | List of AWS regions.                                                                                                                                         |         | yes      |
+| `type`        | `string`       | Cloudwatch service alias (`"alb"`, `"ec2"`, etc) or namespace name (`"AWS/EC2"`, `"AWS/S3"`, etc). See [supported-services][] for a complete list.           |         | yes      |
+| `custom_tags` | `map(string)`  | Custom tags to be added as a list of key / value pairs. When exported to Prometheus format, the label name follows the following format: `custom_tag_{key}`. | `{}`    | no       |
+| `search_tags` | `map(string)`  | List of key / value pairs to use for tag filtering (all must match). Value can be a regex.                                                                   | `{}`    | no       |
+
+[supported-services]: #supported-services-in-discovery-jobs
+
+## static block
+
+The `static` block configures the component to scrape a specific set of CloudWatch metrics. The metrics need to be fully qualified with the following specifications:
+
+1. `namespace`: For example, `AWS/EC2`, `AWS/EBS`, `CoolApp` if it were a custom metric, etc.
+2. `dimensions`: CloudWatch identifies a metric by a set of dimensions, which are essentially label / value pairs. For example, all `AWS/EC2` metrics are identified by the `InstanceId` dimension and the identifier itself.
+3. `metric`: Metric name and statistics.
+
+For example, if you want to scrape the same metrics in the discovery example, but for a specific AWS EC2 instance:
+
+```river
+prometheus.exporter.cloudwatch "static_instances" {
+	sts_region = "us-east-2"
+
+	static "instances" {
+		regions    = ["us-east-2"]
+		namespace  = "AWS/EC2"
+		dimensions = {
+			"InstanceId" = "i01u29u12ue1u2c",
+		}
+
+		metric {
+			name       = "CPUUsage"
+			statistics = ["Sum", "Average"]
+			period     = "1m"
+		}
+	}
+}
+```
+
+You can configure the `static` block one or multiple times to scrape metrics with different sets of `dimensions`.
+
+| Name          | Type           | Description                                                                                                                                                  | Default | Required |
+| ------------- | -------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------ | ------- | -------- |
+| `name`        | `string`       | Identifier of the static scraping job. When exported to Prometheus format corresponds to the `name` label.                                                   |         | yes      |
```

Comment:
I may have misunderstood, but from what I gathered the `name` for `static` blocks is [derived from the block's label](https://github.com/grafana/agent/blob/2787f0faa6dd74aaf3af9df26338077fa11f055e/component/prometheus/exporter/cloudwatch/config.go#L51) (so `static "my_static_block" { }` would have `name == "my_static_block"`). I'm not aware of any other component blocks that do this, and not personally opposed to it, but I think it may be a bit confusing to add it to the arguments of the `static` block as it isn't an argument, per se. Should the description be moved as a note underneath the arguments table? Something like:
```markdown
A `static` block's label corresponds to the `name` label when exported to the Prometheus format.
```

Maybe @clayton-cornell and/or @rfratto can chime in on thoughts for this?

Reply:
Labels for blocks shouldn't be listed as an argument. For an example of what other components do, see the [profile.custom block in pyroscope.scrape](https://grafana.com/docs/agent/latest/flow/reference/components/pyroscope.scrape/#profilecustom-block).

The arguments table should be reserved for literal arguments which are provided by the user via the `ARGUMENT_NAME = ARGUMENT_VALUE` syntax, putting more things in the table (like blocks) might confuse the reader, as we're trying to make the table be indicative of how the user would configure that thing.

Reply:
Corrected following the pyroscope example.